### ISSUE TITLE
NoClassDefFoundError: com/intellij/javascript/nodejs/NodeCoreModulesManager

### ISSUE THREAD

**Issue by gunins** [Fedott/WebStormRequireJsPlugin#59]
Hi
Gotting this error "NoClassDefFoundError: com/intellij/javascript/nodejs/NodeCoreModulesManager"
Since update to Webstorm 11

**Comment by cheahkhing** [comment]
+1, having the same issue, not sure what is the cause, can't seem to get rid of the message

**Comment by bastimeyer** [comment]
@Fedott please fix this :confused:
```
com/intellij/javascript/nodejs/NodeCoreModulesManager
java.lang.NoClassDefFoundError: com/intellij/javascript/nodejs/NodeCoreModulesManager
	at requirejs.Path.probeResolveBasic(Path.java:199)
	at requirejs.Path.resolve(Path.java:102)
	at requirejs.RequirejsProjectComponent.requireResolve(RequirejsProjectComponent.java:657)
	at requirejs.RequirejsReference.resolve(RequirejsReference.java:40)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.checkReferences(JSAnnotatingVisitor.java:274)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.checkReferences(JSAnnotatingVisitor.java:240)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.visitJSLiteralExpression(JSAnnotatingVisitor.java:166)
	at com.intellij.lang.javascript.psi.impl.JSLiteralExpressionImpl.accept(JSLiteralExpressionImpl.java:98)
	at com.intellij.lang.javascript.validation.JSAnnotatingVisitor.annotate(JSAnnotatingVisitor.java:117)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.a(DefaultHighlightVisitor.java:162)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:103)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:354)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$100(GeneralHighlightingPass.java:65)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$3.run(GeneralHighlightingPass.java:283)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:308)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$200(GeneralHighlightingPass.java:65)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$4.run(GeneralHighlightingPass.java:314)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:87)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:311)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:280)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:219)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:82)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:67)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:444)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1169)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:435)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:432)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:206)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)
Caused by: java.lang.ClassNotFoundException: com.intellij.javascript.nodejs.NodeCoreModulesManager PluginClassLoader[com.fedot.idea.requirejs, 0.24]
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:68)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 37 more
```

**Comment by getsnappy** [comment]
+1, having the same issue

**Comment by as19git67** [comment]
Also have the same issue here.

**Comment by prantlf** [comment]
The same here.  Lots of lines:

    21:46:36 NoClassDefFoundError: com/intellij/javascript/nodejs/NodeCoreModulesManager

Apparenly the class was moved to some other package and the detection of the NodeJS core modules has to be updated in `src/requirejs/Path.java`:

    199:    if (NodeCoreModulesManager.isCoreModuleName(this.getPath())) {
                return this.getContainingFile();
            }

**Comment by Fedott** [comment]
Fixed on version 0.25
